### PR TITLE
Prevent the modif. of bookings' dates in the past

### LINKED ISSUE #1753: Prevent the modification of a booking's start and end dates if it is in the past
If a booking has either its start date, end date or both in the past; it should not be allowed to modify them as discussed in the issue #1731.
  - [x] prevent the modification of past start dates on the server side
  - [x] prevent the modification of past start dates on the client side
  - [x] prevent the modification of past end dates on the server side
  - [x] prevent the modification of past end dates on the client side

### PR DESCRIPTION
**Pull request by jacquesd**
- prevents the modification of past start dates on both the server and client
   side
 - prevents the modification of past end dates on both the server and client
   side
 - add the "disabled" functionality to the timerange jquery ui widget
 - fixes #1753

### CODE REVIEW

File: indico/MaKaC/webinterface/tpls/RoomBookingNewBookingPeriodWidget.tpl [hunk 1]
```diff
@@ -92,15 +93,19 @@ ${ form.repeat_interval(type='hidden') }
...
             dateformat: 'dd/mm/yy',
             % if not can_override:
-                minDate: ${ "'{}'".format(min_date.strftime('%d/%m/%Y')) if min_date else 0 },
                 maxDate: ${ room.max_advance_days - 1 if room and room.max_advance_days else 'null' },
             % endif
             showButtonPanel: true,
             changeMonth: true,
             changeYear: true,
-            showOn: 'focus'
+            showOn: 'focus',
```
Comment (ThiefMaster) [inline indico/MaKaC/webinterface/tpls/RoomBookingNewBookingPeriodWidget.tpl]: ...

File: indico/htdocs/js/indico/RoomBooking/validation.js [hunk 2]
```diff
@@ -247,7 +247,10 @@
                         location.href = data.url;
                     } else {
                         var error_box = $('.js-booking-creation-error-box').clone().show();
-                        error_box.find('.js-booking-creation-error-message').text(data.msg);
+                        var errors = error_box.find('.js-booking-creation-error-message');
+                        $.each(data.msg.split('\n'), function addError(_, error) {
+                            errors.append('<li>' + error + '</li>');
```
Comment (ThiefMaster) [inline indico/htdocs/js/indico/RoomBooking/validation.js]: `$('<li>', {text: error})`

File: indico/htdocs/js/indico/jquery/timerange.js [hunk 3]
```diff
@@ -125,6 +125,15 @@
             });
         },
 
+        _setOption: function( key, value ) {
+            if ( key === "disabled" ) {
```
Comment (ThiefMaster) [inline indico/htdocs/js/indico/jquery/timerange.js]: we don't put whitespace inside `()`

**Comment by pferreir** [comment]
Merged in 780a143.

**Pull request closed without merging**